state/distribution_test.go

```diff
@@ -52,16 +54,34 @@ func (s *InstanceDistributorSuite) SetUpTest(c *gc.C) {
 		return &s.distributor, nil
 	}
 
+	a := arch.DefaultArchitecture
+	s.hwChar = &instance.HardwareCharacteristics{
+		Arch: &a,
+	}
+
 	s.wordpress = s.AddTestingApplication(c, "wordpress", s.AddTestingCharm(c, "wordpress"))
 
 	s.machines = make([]*state.Machine, 3)
 	for i := range s.machines {
-		var err error
-		s.machines[i], err = s.State.AddOneMachine(state.MachineTemplate{
-			Series: "quantal",
-			Jobs:   []state.MachineJob{state.JobHostUnits},
+		m, err := s.State.AddOneMachine(state.MachineTemplate{
+			Series:                  "quantal",
+			Jobs:                    []state.MachineJob{state.JobHostUnits},
+			Constraints:             constraints.MustParse("arch=amd64"),
+			HardwareCharacteristics: *s.hwChar,
```

Comment:
Do we need to set it here given that we pass it to `SetProvisioned`?

Reply:
Let me re-evaluate this test as my knowledge has changed since I completed that test.

Reply:
You are indeed correct.